frontends/p4/redundantParsers.h

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2013-present Barefoot Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef _P4_REDUNDANT_PARSERS_H
+#define _P4_REDUNDANT_PARSERS_H
+
+#include "ir/ir.h"
+#include "typeMap.h"
+
+namespace P4 {
+
+/** Find parsers that have an unconditional "accept" in their start
+ *  state, and put them in redundantParsers.
+ */
+class FindRedundantParsers : public Inspector {
+    std::set<cstring> &redundantParsers;
```

Comment:
why string and not IR::P4Parser?

Reply:
In a transform you should use getOriginal(), there are lots of examples in the code